### PR TITLE
feat(router): support configured route in canLoad redirect string

### PR DESCRIPTION
**Pull request by jwx**
# Pull Request

## 📖 Description

This PR makes it possible for a string that's returned from a `canLoad` hook to also be a path matching a configured route or the id of a configured route. (Returning a new `RoutingInstruction` for more complex scenarios is still supported.) As with all string-based navigations, configured routes, if used, are checked before component names.

In addition to the above some minor improvements/clean ups regarding cancellation of an instruction (which is what happens during a redirect) and getting the right navigation content for a viewport have been made. These small refactors don't change any behavior, but instead increase the resilience against race conditions.

### 🎫 Issues

Resolves https://github.com/aurelia/aurelia/issues/1521.

## 📑 Test Plan

- Make tests for the different new redirect combinations
- Make sure existing tests pass

### CODE REVIEW

**Comment by Vheissu** [comment]
Aw yeah @jwx this is great. Thank you.

**Review by Vheissu — APPROVED** [review]

File: packages/router/src/routing-scope.ts [hunk 1]
```diff
@@ -752,26 +761,33 @@ export class RoutingScope {
...
     }
-    return null;
+
+    return found;
   }
 
-  private findMatchingRouteInRoutes(path: string, routes: Route[] | null): FoundRoute | null {
-    if (!Array.isArray(routes)) {
+  private findMatchingRouteInRoutes(path: string, routes: Route[]): FoundRoute | null {
+    if (!Array.isArray(routes) || routes.length === 0) {
```
Comment (bigopon) [inline packages/router/src/routing-scope.ts:790]: `Array.isArray` check can be removed

**Review by bigopon — APPROVED** [review]
Looks good, some of the suggestion can be in the next PRs.

File: packages/router/src/endpoints/viewport.ts [hunk 2]
```diff
@@ -477,23 +484,42 @@ export class Viewport extends Endpoint {
...
-          const canLoadResult = step.previousValue as boolean | LoadInstruction | LoadInstruction[];
+          let canLoadResult = step.previousValue as boolean | LoadInstruction | LoadInstruction[];
           if (typeof canLoadResult === 'boolean') { // canLoadResult: boolean | LoadInstruction | LoadInstruction[],
             if (!canLoadResult) {
               step.cancel();
               coordinator.cancel();
+              this.getNavigationContent(coordinator)!.instruction.nextScopeInstructions = null;
               return;
             }
           } else { // Denied and (probably) redirected
```
[the remark this answers is not in the document]
Comment (jwx) [reply packages/router/src/endpoints/viewport.ts:501]: Yeah.

**Pull request merged**